### PR TITLE
Do not start the web API until the workers are ready

### PR DESCRIPTION
**Pull request by flavio**
Wait for all the workers to be ready before we start serving the web API.

Bootstraping the workers can take some time, depending on their number and the amount of policies each one of them has to initialize.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -130,15 +135,20 @@ fn main() {
     }
 
     let (api_tx, api_rx) = channel::<EvalRequest>(32);
+    let pool_size = matches.value_of("workers").map_or_else(num_cpus::get, |v| {
+        usize::from_str_radix(v, 10).expect("error converting the number of workers")
+    });
+
+    let barrier = Arc::new(Barrier::new(pool_size + 1));
```
Comment (ereslibre) [inline src/main.rs:142]: Didn't know barrier, nice!

File: src/worker.rs [hunk 2]
```diff
@@ -56,16 +61,20 @@ impl WorkerPool {
...
     let mut tx_chans = Vec::<Sender<EvalRequest>>::new();
 
     for n in 1..=size {
       let (tx, rx) = channel::<EvalRequest>(32);
       tx_chans.push(tx);
       let ps = policies.clone();
+      let c = barrier.clone();
 
       thread::spawn(move || {
+        println!("spawining worker {}", n);
```
Comment (ereslibre) [inline src/worker.rs:75]:
```suggestion
        println!("spawning worker {}", n);
```

**Review by ereslibre — APPROVED** [review]
LGTM, thanks @flavio

**Comment by flavio** [comment]
Thanks @ereslibre , I accepted your fix and then I squashed the commits together

**Pull request merged**